### ISSUE TITLE
Does not report any errors

### ISSUE THREAD

**Issue by stellarhoof** [Quramy/typescript-eslint-language-service#49]
I can't seem to get this to work. Is there any way of enabling some sort of debugging?

My `jsconfig.json`:

```json
 "options": {
    "allowJs": true,
    "maxNodeModuleJsDepth": 2,
    "allowSyntheticDefaultImports": true,
    "skipLibCheck": true,
    "noEmit": true,
    "plugins": [
      {
        "name": "typescript-eslint-language-service"
      }
    ]
 }
```

Relevant part of my `tsserver` log:

```log
Info 8    [15:20:10.250] Enabling plugin typescript-eslint-language-service from candidate paths: /Users/berserk/Code/smartprocure/futil/node_modules/typescript/lib/tsserver.js/../../..
Info 9    [15:20:10.251] Loading typescript-eslint-language-service from /Users/berserk/Code/smartprocure/futil/node_modules/typescript/lib/tsserver.js/../../.. (resolved to /Users/berserk/Code/smartprocure/futil/node_modules/node_modules)
Info 10   [15:20:10.630] [typescript-eslint-language-service] config: {"name":"typescript-eslint-language-service"}
Info 11   [15:20:10.640] Plugin validation succeded
Info 12   [15:20:10.658] Starting updateGraphWorker: Project: /Users/berserk/Code/smartprocure/futil/jsconfig.json
Info 13   [15:20:12.965] Finishing updateGraphWorker: Project: /Users/berserk/Code/smartprocure/futil/jsconfig.json Version: 1 structureChanged: true Elapsed: 2307ms
Info 14   [15:20:12.965] Project '/Users/berserk/Code/smartprocure/futil/jsconfig.json' (Configured)
Info 15   [15:20:12.965] 	Files (417)

Info 16   [15:20:12.965] -----------------------------------------------
Info 17   [15:20:13.6] Project '/Users/berserk/Code/smartprocure/futil/jsconfig.json' (Configured)
Info 17   [15:20:13.6] 	Files (417)

Info 17   [15:20:13.6] -----------------------------------------------
Info 17   [15:20:13.6] Open files: 
Info 17   [15:20:13.6] 	FileName: /Users/berserk/Code/smartprocure/futil/src/function.js ProjectRootPath: /Users/berserk/Code/smartprocure/futil
Info 17   [15:20:13.6] 		Projects: /Users/berserk/Code/smartprocure/futil/jsconfig.json
Info 17   [15:20:13.404] [typescript-eslint-language-service] use '@typescript-eslint/parser'
```

My `.eslintrc.js`:

```javascript
module.exports = {
  extends: 'smartprocure',
  parser: 'babel-eslint',
  parserOptions: {
    sourceType: 'module'
  }
}
```

And my `devDependencies`:

```json
{
  "devDependencies": {
    "eslint": "^7.7.0",
    "@typescript-eslint/parser": "^3.9.0",
    "@typescript-eslint/typescript-estree": "^3.9.0",
    "typescript": "^3.9.7",
    "typescript-eslint-language-service": "^3.1.0",
    "typescript-language-server": "^0.4.0",
  }
} 
```

**Comment by mickaelistria** [comment]
Same issue here. Eclipse Wild Web Developer is using this plugin; but trying to upgrade to newer version shows similar symptoms, also with typescript-language-server.

**Comment by lorenzodallavecchia** [comment]
I'm having the same issue. I traced down the cause in my case.

In `tsserver.log` I see.

```
Info 600  [10:49:45.340] [typescript-eslint-language-service] The "path" argument must be of type string. Received undefined
```

The culprit is a call to `path.extname` inside `@eslint/eslintrc/lib/config-array-factory.js` (line 305).

```javascript
function loadConfigFile(filePath) {
    switch (path.extname(filePath)) {
        // ...
    }
}
```

`filePath` is `undefined`.

Tracing back, the missing file path is the real path of `eslint:recommended` and it should have been passed to  the `CascadingConfigArrayFactory` constructor as second-last parameter.

The language service plugin is not passing any parameter (which seems to be allowed by `CascadingConfigArrayFactory`), so the default `undefined` is being used.

In the end, this looks like a ESLint bug, since they have an optional argument with no default. The bug was introduced by the PR eslint/eslintrc#11.

A workaround is to stop using the `eslint:recommended` alias in `.eslintrc.js`.

```javascript
module.exports = {
    // ...
    extends: [
        require.resolve("eslint/conf/eslint-recommended.js"),  // should be "eslint:recommended"
        "plugin:@typescript-eslint/recommended",
    ],
    // ...
};
```

**Comment by mickaelistria** [comment]
FWIW, Wild Web Developer migrated to use the language server that's behind the vscode-eslint extension and doesn't use this tsserver plugin any more.

**Comment by Quramy** [comment]
@lorenzodallavecchia Thanks for your comment. It helped me so much
